addons/promote/common/src/main/java/org/commonjava/indy/promote/validate/PromotionValidationTools.java

```diff
@@ -521,4 +522,35 @@ public Transfer getTransfer( final StoreResource resource )
     {
         return transferManager.getCacheReference( resource );
     }
+
+    public <T> void paralleledEach( Collection<T> collection, groovy.lang.Closure closure )
+    {
+        final Logger logger = LoggerFactory.getLogger( this.getClass() );
+        logger.trace( "Exe parallel on collection {} with closure {}", collection, closure );
+        collection.parallelStream().forEach( e -> {
```

Comment:
Let's use an injected Executor if possible, so we can configure the executor pool in production.